docs/Learning-Environment-Design-Agents.md

```diff
@@ -981,20 +981,45 @@ For an example of how to set up cooperative environments, see the
 Escape Room example environments
 
 #### Cooperative Behaviors Notes and Best Practices
+##### Agents in a Group
+* An Agent can only be registered to one MultiAgentGroup at a time. If you want to re-assign an
+Agent from one group to another, you have to unregister it from the current group first.
 
-* Agents within groups should set the Max Steps parameter the Agent script to 0. Instead,
-handle Max Steps by ending the episode foor the entire Group using `EndGroupEpisode()`, and
-`GroupEpisodeInterrupted()`.
+* Agents within groups should always set the `Max Steps` parameter the Agent script to 0, meaning
+they will never reach a max step. Instead, handle Max Steps with MultiAgentGroup by ending the episode for the entire
+Group using `GroupEpisodeInterrupted()`.
 
-* If an Agent is deactivated in a scene, it must be re-registered to the Multi Agent Group.
+* Agents with different behavior names in the same group are not supported.
+
+* If an Agent finished earlier, e.g. completed tasks/be removed/be killed in the game, do not call
+`EndEpisode()` on the Agent. Instead, disable the Agent and re-enable it when the next episode starts,
+or destroy the agent entirely. Once an agent is disabled, it will not receive any group or
+individual rewards anymore and effectively doesn't exist in the game.
+
+* If an Agent is disabled in a scene, it must be re-registered to the MultiAgentGroup.
+
+* Group rewards are meant to reinforce agents to act in the group's best interest instead of
+individual ones, and are treated differently than individual agent rewards during
+training. So calling AddGroupReward() is not equivalent to calling agent.AddReward() on each agent
+in the group.
 
 * You can still add incremental rewards to Agents using `Agent.AddReward()` if they are
-in a Group. These rewards will only be given to those agents and can only be recieved when the
+in a Group. These rewards will only be given to those agents and can only be received when the
 Agent is active.
 
 * Environments which use Multi Agent Groups can be trained using PPO or SAC, but agents will
 not receive Group Rewards after deactivation/removal, nor will they behave as cooperatively.
 
+##### Ending Episodes for a Group
+* `EndGroupEpisode` and `GroupEpisodeInterrupted` do the same job in the game, but has
+slightly different effect on the training. If the episode is completed, you would want to call
+`EndGroupEpisode`. But if the episode is not over but it has been running for enough steps, i.e.
+reaching max step, you would call `GroupEpisodeInterrupted`.
+
+##### Multiple Groups:
+* You can have multiple MultiAgentGroup in one training field, but make sure they call
+`EndGroupEpisode()` and `GroupEpisodeInterrupted` at the same time and their episodes are in sync.
```

Comment:
It's OK for them to be out of sync right? e.g. if they're different behavior names or different groups

Reply:
I'll remove that